vmdb/spec/models/miq_enterprise_spec.rb

```diff
@@ -0,0 +1,76 @@
+require "spec_helper"
+
+describe MiqEnterprise do
+  before(:each) { @ent = FactoryGirl.create(:miq_enterprise) }
+
+  it "#miq_regions=" do
+    expect(@ent.miq_regions.size).to eq(1)
+  end
+
+  it "#ext_management_systems=" do
+    (1..5).each { |i| FactoryGirl.create(:ems_vmware, :name => "EMS_#{i}") }
+
+    expect(@ent.ext_management_systems.size).to eq(5)
+  end
+
+  it "#vms_and_templates=" do
+    ems  = FactoryGirl.create(:ems_vmware, :name => "EMS")
+    host = FactoryGirl.create(:host,       :name => "Host")
+    vm_1 = FactoryGirl.create(:vm_vmware,  :host => host, :name => "EMS ID VM 1",  :uid_ems => "1", :ems_id => ems.id)
+    vm_2 = FactoryGirl.create(:vm_vmware,  :host => host, :name => "EMS ID VM 2",  :uid_ems => "2", :ems_id => ems.id)
+    vm_3 = FactoryGirl.create(:vm_vmware,  :host => host, :name => "No EMS ID VM", :uid_ems => "3")
+
+    FactoryGirl.create(:hardware, :vm_or_template_id => vm_1.id, :host => host)
+    FactoryGirl.create(:hardware, :vm_or_template_id => vm_2.id, :host => host)
+    FactoryGirl.create(:hardware, :vm_or_template_id => vm_3.id, :host => host)
+
+    expect(@ent.vms_and_templates.size).to eq(2)
+  end
+
+  it "#vms=" do
+    FactoryGirl.create(:vm_vmware, :vendor => "redhat", :name => "VMWare",  :ems_id => nil)
+    FactoryGirl.create(:vm_redhat, :vendor => "redhat", :name => "Red Hat", :ems_id => 10000000000001)
+    FactoryGirl.create(:vm_amazon, :vendor => "amazon", :name => "Amazon",  :ems_id => 10000000000002)
+
+    expect(@ent.vms.size).to eq(2)
+  end
+
+  it "#miq_templates=" do
+    ems_rh = FactoryGirl.create(:ems_redhat, :name => "Red Hat")
+    ems_vm = FactoryGirl.create(:ems_vmware, :name => "VMWare")
+
+    FactoryGirl.create(:miq_template, :name => "Red Hat", :location => "Someplace", :vendor => "redhat", :ems_id => ems_rh.id)
+    FactoryGirl.create(:miq_template, :name => "Vm Ware", :location => "Noplace",   :vendor => "vmware", :ems_id => ems_vm.id)
+
+    (1..3).each { |i| FactoryGirl.create(:miq_template, :name => "Template_#{i}", :location => "Someplace", :vendor => "amazon") }
+
+    expect(@ent.miq_templates.size).to eq(2)
+  end
+
+  it "#hosts=" do
+    @ems_1 = FactoryGirl.create(:ems_vmware)
+    FactoryGirl.create(:host_vmware, :ems_id => @ems_1.id)
+    @ems_2 = FactoryGirl.create(:ems_redhat)
+    FactoryGirl.create(:host_redhat, :ems_id => @ems_2.id)
+    @ems_3 = FactoryGirl.create(:ems_openstack_infra)
+    FactoryGirl.create(:host_vmware_esx, :ems_id => @ems_3.id)
+
+    expect(@ent.hosts.size).to eq(3)
+  end
+
+  it "#storages=" do
+    FactoryGirl.create(:storage, :name => "test_storage_vmfs",  :store_type => "VMFS")
+    FactoryGirl.create(:storage, :name => "test_storage_nfs_1", :store_type => "NFS1")
+    FactoryGirl.create(:storage, :name => "test_storage_nfs_2", :store_type => "NFS2")
+    FactoryGirl.create(:storage, :name => "test_storage_foo",   :store_type => "FOO")
+
+    expect(@ent.storages.size).to eq(4)
+  end
+
+  it "#policy_events=" do
```

Comment:
This (and all the other `it` statements above) should match the method name, and thus should NOT have the trailing `=` sign.